config/v1/types_infrastructure.go

```diff
@@ -678,6 +689,23 @@ type AlibabaCloudResourceTag struct {
 	Value string `json:"value"`
 }
 
+// NutanixPlatformSpec holds the desired state of the Nutanix infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type NutanixPlatformSpec struct{}
+
+// NutanixPlatformStatus holds the current status of the Nutanix infrastructure provider.
+type NutanixPlatformStatus struct {
+	// apiServerInternalIP is an IP address to contact the Kubernetes API server that can be used
+	// by components inside the cluster, like kubelets using the infrastructure rather
+	// than Kubernetes networking. It is the IP that the Infrastructure.status.apiServerInternalURI
+	// points to. It is the IP for a self-hosted load balancer in front of the API servers.
+	APIServerInternalIP string `json:"apiServerInternalIP,omitempty"`
```

Comment:
why do we need this in addition?

Reply:
@sttts AFAIK on-prem platforms are expected to provide both these fields to the machine config operator. see: https://github.com/openshift/machine-config-operator/pull/2852/files#